aten/src/ATen/native/mkldnn/Pooling.cpp

```diff
@@ -144,6 +192,78 @@ static Tensor _mkldnn_pool2d(
   return new_with_itensor_mkldnn(std::move(y), input.options());
 }
 
+static Tensor _mkldnn_pool2d_backward(
+    const Tensor& grad_output,
+    const Tensor& output,
+    const Tensor& input,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    bool ceil_mode,
+    ideep::algorithm algo) {
+
+  auto kernel_size_vec = expand_param_if_needed(kernel_size, "kernel_size", 2);
+  auto stride_vec = expand_param_if_needed(stride, "stride", 2);
+  auto padding_vec = expand_param_if_needed(padding, "padding", 2);
+  auto padding_vec_l = padding_vec;
+  auto padding_vec_r = padding_vec;
+  auto dilation_vec = expand_param_if_needed(dilation, "dilation", 2);
+
+  if (ceil_mode) {
+    // MKLDNN does not support ceil mode, so we adjust padding
+    // on the right side to match behavior. Adjust output size
+    // accordingly.
+    const std::vector<int64_t> output_sizes_ceil = pool_output_sizes(
+        input.sizes(),
+        kernel_size_vec,
+        stride_vec,
+        padding_vec_l,
+        padding_vec_r,
+        dilation_vec,
+        true /* ceil_mode */);
+
+    // adjust padding until output sizes agree
+    bool all_equal = false;
+    std::vector<int64_t> output_sizes;
+    while (!all_equal) {
```

Comment:
Given that all ops are calling here. How expensive is this? Should there be a warning in the docs that the ceil_mode might be inefficient for mkldnn Tensors?